app-kotlin/build.gradle

```diff
@@ -5,37 +5,46 @@ buildscript {
   }
 
   dependencies {
-    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.21'
+    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.0'
   }
 }
 
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 
 android {
-  compileSdkVersion 28
-  buildToolsVersion '28.0.3'
+  compileSdkVersion 31
 
-  defaultConfig {
+    defaultConfig {
     applicationId 'au.com.gridstone.trainingkotlin'
     minSdkVersion 21
-    targetSdkVersion 28
+    targetSdkVersion 31
     versionCode 1
     versionName '1.0'
   }
 
   buildTypes {
     release {
       minifyEnabled false
+      proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
     }
   }
 
-  sourceSets {
-    main.java.srcDirs += 'src/main/kotlin'
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
+  kotlinOptions {
+    jvmTarget = '1.8'
   }
 }
 
 dependencies {
-  implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.3.21'
-  implementation 'androidx.appcompat:appcompat:1.0.2'
+  implementation 'androidx.core:core-ktx:1.7.0'
+  implementation 'androidx.appcompat:appcompat:1.3.0'
+  implementation 'com.google.android.material:material:1.4.0'
+  implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+  testImplementation 'junit:junit:4.13.2'
+  androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+  androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
```

Comment:
do we need all these dependencies?

Reply:
Not sure but might needed in future i will remove for now and added when required